packages/colorpickers/src/elements/ColorSwatch/index.tsx

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { forwardRef, useContext } from 'react';
+import PropTypes from 'prop-types';
+import { ThemeContext } from 'styled-components';
+import { Tooltip } from '@zendeskgarden/react-tooltips';
+import { useGrid } from '@zendeskgarden/container-grid';
+import { useId } from '@zendeskgarden/container-utilities';
+import CheckIcon from '@zendeskgarden/svg-icons/src/12/check-sm-fill.svg';
+import { StyledSwatchButton, StyledIcon, StyledCell, StyledColorSwatch } from '../../styled';
+
+export interface ILabeledColor {
+  value: string;
+  label: string;
+}
+
+export interface IColorSwatchProps {
+  /** Sets the two-dimension array of labeled HEX and RGB/A string colors */
+  colors: ILabeledColor[][];
+  /** Sets the focused row index in a controlled color swatch */
+  rowIndex?: number;
+  /** Sets the focused column index in a controlled color swatch */
+  colIndex?: number;
+  /** Sets the selected row index in a controlled color swatch */
+  selectedRowIndex?: number;
+  /** Sets the selected column index in a controlled color swatch */
+  selectedColIndex?: number;
+  /** Sets the default focused row index in an uncontrolled color swatch */
+  defaultRowIndex?: number;
+  /** Sets the default focused column index in an uncontrolled color swatch */
+  defaultColIndex?: number;
+  /** Sets the default selected row index in an uncontrolled color swatch */
+  defaultSelectedRowIndex?: number;
+  /** Sets the default selected column index in an uncontrolled color swatch */
+  defaultSelectedColIndex?: number;
+  /** Handles color swatch changes */
+  onChange?: (rowIndex: number, colIndex: number) => void;
+  /** Handles color swatch select event */
+  onSelect?: (rowIndex: number, colIndex: number) => void;
+}
+
+/**
+ * @extends HTMLAttributes<HTMLTableElement>
+ */
+export const ColorSwatch = forwardRef<HTMLTableElement, IColorSwatchProps>(
+  ({ colors, ...props }, ref) => {
+    const { rtl } = useContext(ThemeContext);
+    const { getGridCellProps } = useGrid({
+      rtl,
+      matrix: colors,
+      selection: true,
+      wrap: true,
+      idPrefix: useId(),
+      ...props
+    });
+
+    return (
+      <StyledColorSwatch ref={ref}>
+        <tbody>
+          {colors.map((row: ILabeledColor[], rowIdx: number) => (
+            <tr key={row[0].value}>
+              {row.map((color: ILabeledColor, colIdx: number) => {
+                const { label, value } = color;
+                const { 'aria-selected': ariaSelected, ...other } = getGridCellProps({
+                  colIdx,
+                  rowIdx,
+                  type: 'button',
+                  role: undefined
```

Comment:
@exelarios this is belt-and-suspenders programming. We build an accessibility hook that doesn't make assumptions re: the ability to use semantic markup. But on this side, we follow the [first rule of ARIA](https://www.w3.org/TR/using-aria/#rule1) by using an element type that subsumes the `role`, making it redundant.

Reply:
Normally `role` is `gridcell` but since presumably the semantic element `td` was used it implies `role="gridcell"` inherently IIRC. https://www.w3.org/TR/wai-aria-1.1/#gridcel
<img width="459" alt="Screen Shot 2021-09-30 at 11 59 08 AM" src="https://user-images.githubusercontent.com/1975639/135514789-3f30fb9c-5266-4cbb-afb1-be051515bb8d.png">
Im starting to think about this like a UML Object Oriented Inheritance tree.

Reply:
Oh wow! thanks for the insight and the links to point me in the right direction. I gotta review the w3, whenever I get the chance.

`belt-and-suspenders programming`, well that's a new one I haven't heard of. 😆